### ISSUE TITLE
[Question] Install Gymnasium with Mujoco on M1

### ISSUE THREAD

**Issue by hg6185** [Farama-Foundation/Gymnasium#1020]
### Question

Hello, 

maybe this is a dumb question, but how can I install Gymnasium with Mujoco Support on an M1.

When I simply install Gymnasium[all] I get a message that I am missing mujoco under ```./mujoco/mujoco-210``` .
When I add a mujoco installation, I am getting the message that GCC9.X is missing. 
```
RuntimeError: Could not find supported GCC executable.

HINT: On OS X, install GCC 9.x with `brew install gcc@9`. or `port install gcc9`.
```
I tried everything, but I don't seem to get it running. Is the problem still the clang gcc problem?
With home-brew I installed gcc to the following path ~/opt/homebrew/bin/gcc-11

Thank you for the support. It is much appreciated!

**Comment by Kallinteris-Andreas** [comment]
Just install the components you want `pip install gymnasium[box2d, mujoco, other]`

I can not really help with installing `mujoco-py` in ARM MACs, all I can suggest
1) try using the X86 version 
2) use gcc9 as suggested,

**Comment by hg6185** [comment]
Thank you for your answer. 

Unfortunately is gcc9 not longer available due to upstream incompatibilities. Which is why I mapped the gcc-9 commands to GCC. However this did not lead to to success. Maybe someone else will encounter the same issue and is more advanced in this compiler stuff :D

**Comment by pseudo-rnd-thoughts** [comment]
The issue is related to `mujoco-py` (the v2 and v3 mujoco environments) so I would try `gymnasium[mujoco]` which only supports `v4` and soon `v5` environments

**Comment by Kallinteris-Andreas** [comment]
@pseudo-rnd-thoughts  I Believe you have an Apple laptop Have you gotten `MuJoCo-py` Pi to work in your computer?

Should we remove MuJoCo from `pip install gymnasium[all]`, It seems to cause issues to users who do not intend to use old versions of MuJoCo environments?

**Comment by pseudo-rnd-thoughts** [comment]
I have an old intel MacBook so I can't test for M1 
Personally, I don't think we should remove mujoco py from all

**Comment by hg6185** [comment]
Unfortunately, I doesn't – Maybe something else is wrong set-up in my computer. Mujoco with it's python binding works like a charm. 
Today, I tried it again in a new env, but I still ran into the following problems. If I create the installation at .mujoco/mucoco210, I still get the gcc error.

**Comment by pseudo-rnd-thoughts** [comment]
@hg6185 Can you confirm is the issue only for `pip install gymnasium[mujoco]` or `pip install gymnasium[mujoco-py]`?
If it is for mujoco-py, are you certain that you need the v2 or v3 environments?

**Comment by hg6185** [comment]
No, I cannot confirm this. The issue occurs also in v4 envs. For instance, If I run 
```import gymnasium as gym
gym.make("Pusher-v4") ```
I get the following error:
```
RuntimeError: Could not find supported GCC executable.

HINT: On OS X, install GCC 9.x with `brew install gcc@9`. or `port install gcc9`.
```

The issue is related to the script that searches for the cpp compiler. However, whatever I do with the path variables, the GCC is not accepted by the mujoco simulator in .mujoco/mujoco210. Is it not possible to change this? Normal mujoco and dm_control works with my setup?

**Comment by Kallinteris-Andreas** [comment]
@hg6185 `pip uninstall mujoco-py` and MuJoCo environments v4 and v5 should work

**Comment by hg6185** [comment]
YESSSS!!!! It workssss thank you so much!

(I also downgraded to mujoco==2.3.0)
